pkg/client/typed_client.go

```diff
@@ -108,24 +111,45 @@ func (c *typedClient) DeleteAllOf(ctx context.Context, obj Object, opts ...Delet
 		Error()
 }
 
+// Sets the name and namespace for the returned object if an ApplyConfiguration was supplied
+func setNameNamespace(obj Object, ac interface{}) {
```

Comment:
I can add some additional checks for invalid structs, but I think this is the best way to copy over the fields without explicitly modifying k/k in 1.22 to declare some interface.